common/src/BasePlugin.php

```diff
@@ -116,30 +116,59 @@ public function getProductMetaKeys(): array {
             WHERE
                 p.post_type = 'product'
                 AND pm.meta_key <> ''
+                AND pm.meta_value <> ''
         ");
         return array_map(
             function ($value) {
                 return [
                     'type' => 'meta_key',
-                    'name' => $value
+                    'name' => $value,
+                    'example_value' => substr($this->getMetaValue($value), 0, 15),
                 ];
             },
             $meta_keys
         );
     }
 
-    public function getProductKeys() {
+    public function getSelectOptions(string $selected_key, bool $suggested = false) {
+        $options = [];
+        foreach ($this->getProductKeys() as $key) {
+            if ($key['suggested'] != $suggested) {
+                continue;
+            }
+            $options[] = '<option value="' . $key['option_value'] . '" '
+                . ($key['option_value'] == $selected_key ? 'selected' : '') . '>'
+                . $key['label']
+                . '</option>';
```

Comment:
You forgot to apply htmlentities on example_value.  I always apply htmlentities at the point where you are generating the HTML, as you won't have to remember which values are safe and which are not.  So I would ask you to apply htmlentities here to all the value and label, and remove it on lines 151-153.

Reply:
Also sprintf would be a lot more readable here.

~~~
sprintf(
  '<option value="%s"%s>%s</option>',
  htmlentities($key['option_value']),
  $key['option_value'] === $selected_key ? ' selected' : '',
  htmlentities($key['label'])
);
~~~

Reply:
pff `sprintf` is perfect here